mne/io/edf/edf.py

```diff
@@ -114,6 +114,11 @@ def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
         annotmap = self._raw_extras[fi]['annotmap']
         subtype = self._raw_extras[fi]['subtype']
 
+        # Interpolating stim chan except it later gets overwritten.
+        if ((not (annot and annotmap or tal_channel is not None) and
+                (stim_channel in sel))):
+            warn('Interpolating stim channel. Events may jitter.')
```

Comment:
Actually instead of just being outside the loop, thinking about how this code actually works it should probably be in the EDF info reading section. Otherwise you can still get this warning a lot if you don't preload data.